### ISSUE TITLE
Shape inference fails on the Split operator

### ISSUE THREAD

**Issue by zhengly123** [onnx/onnx#3358]
# Bug Report

### Describe the bug
Shape inference fails on the Split operator without error or warning messages.
This is the shape inference result: 
![image](https://user-images.githubusercontent.com/22888563/112264064-47fd2200-8cab-11eb-98ef-71eef85081be.png)

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*): Debian 10
- ONNX version (*e.g. 1.7*):  1.8.1
- Python version: 3.7.3

### Reproduction instructions
- Describe the code to reproduce the behavior.
```
import functools
import sys
import onnx
import onnx.shape_inference

def import_onnx(net: str):
    model = onnx.load(net)
    infered_model = onnx.shape_inference.infer_shapes(model)
    onnx.save(infered_model, sys.argv[1][:-5]+'.inferred.onnx' )
import_onnx(sys.argv[1])
...
```
- Attach the ONNX model to the issue (where applicable)

[split.inferred.zip](https://github.com/onnx/onnx/files/6194827/split.inferred.zip)

### Expected behavior
The shapes of tensors after the Split OP should be inferred.

**Comment by jcwchen** [comment]
Hi @zhengly123,
The result of shape inference seems correct from my end with the provided model. You need to check them in `model.graph.value_info`. Could you take a look at it? Thanks.

**Comment by zhengly123** [comment]
> Hi @zhengly123,
> The result of shape inference seems correct from my end with the provided model. You need to check them in `model.graph.value_info`. Could you take a look at it? Thanks.

Thanks for your quick reply.
I print `model.graph.value_info`. But this output shows the shapes of outputs of Split and some following tensors are missing. What's more, I can repro this problem on another server with the same version of onnx. 
```
def import_onnx(net: str):
    model = onnx.load(net)
    infered_model = onnx.shape_inference.infer_shapes(model)
    print(infered_model.graph.value_info)
```


```
// Tensors with shape info are omitted
, name: "51"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_value: 1
      }
      dim {
        dim_value: 176
      }
      dim {
        dim_value: 38
      }
      dim {
        dim_value: 38
      }
    }
  }
}
, name: "52"
type {
  tensor_type {
    elem_type: 1
  }
}
, name: "53"
type {
  tensor_type {
    elem_type: 1
  }
}
, name: "54"
type {
  tensor_type {
    elem_type: 1
  }
}
, name: "65"
type {
  tensor_type {
    elem_type: 1
  }
}
, name: "57"
type {
  tensor_type {
    elem_type: 1
  }
}
, name: "68"
type {
  tensor_type {
    elem_type: 1
  }
}
, name: "60"
type {
  tensor_type {
    elem_type: 1
  }
}
]
```

**Comment by jcwchen** [comment]
I believe this bug has been fixed by https://github.com/onnx/onnx/pull/2549. Since it is not included in 1.8.1, could you try the latest ONNX? A quick experiment is using the onnx-weekly package: `pip install -i https://test.pypi.org/simple/ onnx-weekly`.

**Comment by zhengly123** [comment]
The onnx-weekly works on it! Thanks.
